StreamChatClient_v3/WebSocketClient/WebSocketClient.swift

```diff
@@ -87,16 +98,25 @@ class WebSocketClient {
     /// An object containing external dependencies of `WebSocketClient`
     private let environment: Environment
     
-    init(
-        urlRequest: URLRequest,
-        sessionConfiguration: URLSessionConfiguration,
-        eventDecoder: AnyEventDecoder,
-        eventMiddlewares: [EventMiddleware],
-        reconnectionStrategy: WebSocketClientReconnectionStrategy = DefaultReconnectionStrategy(),
-        environment: Environment = .init()
-    ) {
+    private func engineBuilder() throws -> WebSocketEngine {
```

Comment:
It's not a builder. It returns `WebSocketEngine`

Reply:
Not sure what you mean here. It's called `builder` to signalize that it creates a new instance, that it's not some existing object.

Reply:
So it doesn't return a builder it _creates_ an instance, then it should be renamed to `createEngine` as you do in other places.

Reply:
It's actually `builder` everywhere, but if you prefer `createEngine` I have no problem with that.

Reply:
The function named `engineBuilder()` and I think it's clear that I expect it will return a builder and then we can use the builder to create engines. But the content of the function is different. It creates an engine.
You on the same time in other places user prefix `create` for cases like that and it's not clear why you decide to change it here.

Reply:
fixed